sites/docs/src/content/docs/contributing/nf-test/assertions.md

```diff
@@ -368,6 +368,29 @@ Therefore, even if you explicitly exclude the file during the `endFileRercurse`
 Therefore, by excluding directories, you do not get an accidental 'double' listing of files you wish to exclude.
 :::
 
+### Snapshotting variable binary files with file size
+
+_Context_: You have a tool that always produces a binary file that cannot be asserted for valid contents such as a string.
+
+_Motivation_: You want to be able to still check that the binary file contains 'something' rather than just the existence of the file.
+
+To compare an exact file size (in bytes)
+
+```nextflow
+"malt/malt_index/ref.idx - correct file size: ${file("$outputDir/malt/malt_index/ref.idx").length()}",
+```
+
+To check for a minimum size (in bytes)
+
+```nextflow
+"malt/malt_index/ref.idx - minimum file size: ${file("$outputDir/malt/malt_index/ref.idx").length() >= 61616}",
+```
+
+_Explanation_: When you have a binary file that can have variable contents, you cannot check an md5sum (variability) the contents for plain text strings (as it's a binary file).
+
+While you could check simply for the existence of a file, it maybe some tools can produce binary files that have 'insufficent' contents for it to work.
```

Comment:
```suggestion
While you could check simply for the existence of a file, it may be that some tools can produce binary files that have 'insufficent' contents for it to work.
```